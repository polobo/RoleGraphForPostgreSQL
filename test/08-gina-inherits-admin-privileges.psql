-- ADMIN OPTION for granting object permissions (not roles...) applies through inheritance as well as set
-- Gina the Fixed-Ops Manager can administer both technicians and managers
-- Manager is immediately inherited
-- Techicians requires a set to advisors first but then it is inherited
\if :is_su
    SELECT format('CREATE USER u8_gina_%s;', :'ctx') \gexec
    SELECT format('CREATE GROUP g8a_general_manager_%s;', :'ctx') \gexec
    SELECT format('CREATE GROUP g8b_fixedops_manager_%s;', :'ctx') \gexec
    SELECT format('CREATE GROUP g8c_service_manager_%s;', :'ctx') \gexec
    SELECT format('CREATE GROUP g8d_service_advisor_%s;', :'ctx') \gexec
    SELECT format('CREATE GROUP g8e_service_tech_%s;', :'ctx') \gexec
    SELECT format('GRANT g8a_general_manager_%1$s TO u8_gina_%1$s WITH SET FALSE;', :'ctx') \gexec
    SELECT format('GRANT g8b_fixedops_manager_%1$s TO g8a_general_manager_%1$s WITH SET FALSE, ADMIN OPTION;', :'ctx') \gexec
    SELECT format('GRANT g8c_service_manager_%1$s TO g8b_fixedops_manager_%1$s WITH SET FALSE;', :'ctx') \gexec
    SELECT format('GRANT g8d_service_advisor_%1$s TO u8_gina_%1$s WITH INHERIT FALSE;', :'ctx') \gexec
    SELECT format('GRANT g8e_service_tech_%1$s TO g8d_service_advisor_%1$s WITH SET FALSE, ADMIN OPTION;', :'ctx') \gexec
\endif
