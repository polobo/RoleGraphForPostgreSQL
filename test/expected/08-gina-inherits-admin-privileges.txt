                                                                                                                                Test Result for 08-gina-inherits-admin-privileges
         rolname         |                         administration                          | memberof_users |                           memberof_groups                            |                       member_users                       |                          member_groups                           
-------------------------+-----------------------------------------------------------------+----------------+----------------------------------------------------------------------+----------------------------------------------------------+------------------------------------------------------------------
 g8a_general_manager_su  | of g8b_fixedops_manager_su from superuser                       |                | g8b_fixedops_manager_su from superuser (i)                          +| u8_gina_su from superuser (i)                            | 
                         |                                                                 |                | g8c_service_manager_su via g8b_fixedops_manager_su/superuser (i)     |                                                          | 
 g8b_fixedops_manager_su | by u8_gina_su via g8a_general_manager_su/superuser             +|                | g8c_service_manager_su from superuser (i)                            | u8_gina_su via g8a_general_manager_su/superuser (i)      | g8a_general_manager_su from superuser (i)
                         | by g8a_general_manager_su from superuser                        |                |                                                                      |                                                          | 
 g8c_service_manager_su  |                                                                 |                |                                                                      | u8_gina_su via g8b_fixedops_manager_su/superuser (i)[+1] | g8a_general_manager_su via g8b_fixedops_manager_su/superuser (i)+
                         |                                                                 |                |                                                                      |                                                          | g8b_fixedops_manager_su from superuser (i)
 g8d_service_advisor_su  | of g8e_service_tech_su from superuser                           |                | g8e_service_tech_su from superuser (i)                               | u8_gina_su from superuser (s)                            | 
 g8e_service_tech_su     | by u8_gina_su via g8d_service_advisor_su/superuser             +|                |                                                                      | u8_gina_su via g8d_service_advisor_su/superuser (i*)     | g8d_service_advisor_su from superuser (i)
                         | by g8d_service_advisor_su from superuser                        |                |                                                                      |                                                          | 
 cr_admin                | Create Role                                                     |                |                                                                      |                                                          | 
 u8_gina_su              | of g8b_fixedops_manager_su via g8a_general_manager_su/superuser+|                | g8a_general_manager_su from superuser (i)                           +|                                                          | 
                         | of g8e_service_tech_su via g8d_service_advisor_su/superuser     |                | g8b_fixedops_manager_su via g8a_general_manager_su/superuser (i)    +|                                                          | 
                         |                                                                 |                | g8c_service_manager_su via g8b_fixedops_manager_su/superuser (i)[+1]+|                                                          | 
                         |                                                                 |                | g8d_service_advisor_su from superuser (s)                           +|                                                          | 
                         |                                                                 |                | g8e_service_tech_su via g8d_service_advisor_su/superuser (i*)        |                                                          | 
(7 rows)

